bentoml/server/gunicorn/__init__.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Atalaya Tech, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from dependency_injector.wiring import inject, Provide
+
+from bentoml.configuration.containers import BentoMLContainer
+
+logger = logging.getLogger(__name__)
+
+
+@inject
```

Comment:
instead of moving those modules around, could we do something like this:

```
def start_prod_server(...)
   ...

if psutil.POSIX:
   start_prod_server = inject(start_prod_server)
```

Or:

```
if psutil.POSIX:
   @inject
    def start_prod_server(...)
          ...
else:
    def start_prod_server(...)
            raise XXXException("prod server only available under POSIX system")
```

Reply:
This wouldn't work unfortunately because `wire` operation does a `pkgutil.walk_packages` and it imports the modules even if they are imported within a function def. In this case here, Gunicorn modules will be imported when we `container.wire(server)`.

https://github.com/ets-labs/python-dependency-injector/blob/d3720bd6dd074677e6d07225ff80ba402d5ac117/src/dependency_injector/wiring.py#L439

Reply:
@ssheng this linked code only imports the packages with the given prefix, e.g. 'bentoml.server.', right?  If the import of gunicorn library is conditional based on the platform information, `pkgutil.walk_packages` should not be importing gunicorn right?

Reply:
If I understood correctly. `module = importlib.import_module(module_info.name)` is recursive.